### DISCUSSION TITLE
Existing surface code development

### DISCUSSION THREAD

**Discussion by quantumjim** [qiskit-community/qiskit-qec#4] [General]
I'm not sure how it would fit with what is here, but note that some Qiskit advocates have been working on integrating surface codes into Qiskit Ignis
https://github.com/qiskit-community/topological_codes

Of course it won't go onto Ignis now, but I was thinking to host their project on qiskit-community.

**Comment by dsvandet** [comment]
I would certainly be up for including code or ideas that would satisfy the level of generality that we are after or form simply showing how surface codes can be handled in Qiskit. We are currently focusing on designing and developing a framework that (as a start) supports working with subsystems codes (abstractly and experimentally (logical or circuit)) and having the ability to "swap" out different error decoders or noise models etc. and to enable gauge fixing (hence code deformations and lattice surgery)  etc. Now as the design of the framework is at the beginning it would be great to better understand what has been done so far.
